src/cpu/flexcpu/simple_dataflow_cpu.cc

```diff
@@ -204,17 +206,19 @@ SimpleDataflowCPU::requestBranchPredictor(
 
 void
 SimpleDataflowCPU::requestDataAddrTranslation(const RequestPtr& req,
-    ThreadContext* tc, bool write,
+    ThreadContext* tc, bool write, std::weak_ptr<ExecContext> context,
```

Comment:
This is where I think a dedicated management block might be useful, since we're creating a typed weak_ptr without actually using the type at all.

A better alternative in this specific case I think is std::weak_ptr<void>, because the communicates that we're using the weak_ptr aspect of it to control the request, and we don't care at all what the pointer points to because we don't use it.